doc/src/build/dev_cheat_sheet.md

```diff
@@ -34,8 +33,10 @@ Quick reference on best practices for Sui Network developers.
 
 # Apps
 
-- Apps should use the wallet's [`signTransactionBlock`](https://sui-wallet-kit.vercel.app/) API, then submit the transaction via a call to [`execute_transactionBlock`](https://docs.sui.io/sui-jsonrpc#sui_executeTransactionBlock) on the app's full node, *not* use the wallet's `signAndExecuteTransactionBlock` API. This ensures read-after-write-consistency--reads from the app's full node will reflect writes from the transaction right away instead of waiting for a checkpoint.
-- Whenever possible, use [programmable transaction blocks](https://www.notion.so/Programmable-Transactions-4264a0fb3034416bba5f8b6dec288b19) to compose existing on-chain functionality rather than publishing new smart contract code.
+- For optimal performance and data consistency, apps should submit writes and reads for the same full node. In the TS SDK, this means that apps should use the wallet's [`signTransactionBlock`](https://sui-wallet-kit.vercel.app/) API, then submit the transaction via a call to [`execute_transactionBlock`](https://docs.sui.io/sui-jsonrpc#sui_executeTransactionBlock) on the app's full node, *not* use the wallet's `signAndExecuteTransactionBlock` API. This ensures read-after-write-consistency--reads from the app's full node will reflect writes from the transaction right away instead of waiting for a checkpoint.
+- For lower latency, use [`executeTransactionBlock`](https://docs.sui.io/sui-jsonrpc#sui_executeTransactionBlock) with `"showEffects": false` and `"showEvents": false` if your app needs to know that a transaction was confirmed, but does not immediately need to see the transaction effects or read the objects/events written by the transaction.
+- Apps should implement a local cache for frequently read data rather than over-fetching from the full node.
+- Whenever possible, use [programmable transaction blocks](https://www.notion.so/Programmable-Transactions-4264a0fb3034416bba5f8b6dec288b19) to compose existing on-chain functionality rather than publishing new smart contract code. Programmable transaction blocks allow large-scale batching and heterogenous composition, driving already-low gas fees down even further.
```

Comment:
Just noticed that this is a link to an internal notion, we should link to the public docs instead.

```suggestion
- Whenever possible, use [programmable transaction blocks](https://docs.sui.io/build/prog-trans-ts-sdk) to compose existing on-chain functionality rather than publishing new smart contract code. Programmable transaction blocks allow large-scale batching and heterogenous composition, driving already-low gas fees down even further.
```